### ISSUE TITLE
Will it be possible to add an automatic  LM Studio start on system startup functionality ?

### ISSUE THREAD

**Issue by ghevge** [lmstudio-ai/lmstudio-bug-tracker#78]
Will it be possible to add an automatic LM Studio start on system startup functionality ?

Also it will be nice to have a functionality to automatically load a specified model on startup too. 

Thanks

**Comment by YorkieDev** [comment]
@ghevge I'm not familiar with the process on Linux, but in Windows you do the following:

Bring up the run window and type: shell:appsfolder and look for the LM Studio shortcut, then do the same to search for shell:startup then just drag the LM Studio shortcut from the appsfolder to the startup folder, then LM Studio will automatically start on startup.

**Comment by Hafer0** [comment]
@YorkieDev that doesn't actually start the server, only the UI of LM Studio

**Comment by PelleRavn** [comment]
I agree, it would be awesome if it could just start the local server automatically when the application starts

**Comment by funmonty** [comment]
Yes need this feature to run lms on startup using a particular model and run the server

**Comment by yagil** [comment]
This is now in beta: https://lmstudio.ai/beta-releases
Docs: https://lmstudio.ai/docs/advanced/headless

**Comment by mhagnumdw** [comment]
> This is now in beta: https://lmstudio.ai/beta-releases
> Docs: https://lmstudio.ai/docs/advanced/headless

It seems that it is not yet possible to run on a Linux server without a graphical interface, right? Because you must first enable the "Local LLM Service (headless)" option in the LM Studio GUI.

**Comment by YorkieDev** [comment]
@mhagnumdw for your usecase it's probably best to run raw llama.cpp (one of the engines behind LM Studio): https://github.com/ggerganov/llama.cpp

**Comment by yagil** [comment]
> > This is now in beta: https://lmstudio.ai/beta-releases
> > Docs: https://lmstudio.ai/docs/advanced/headless
> 
> It seems that it is not yet possible to run on a Linux server without a graphical interface, right? Because you must first enable the "Local LLM Service (headless)" option in the LM Studio GUI.

2 part answer:

1. you don’t need to interact with the GUI even on Linux. That’s because you can simply add `lms server start` to your systemd start up service

2. you still need GUI dependencies to exist on your system. We will provide a solution for this soon (the are workarounds but they’re not very pretty)

**Comment by PelleRavn** [comment]
> This is now in beta: https://lmstudio.ai/beta-releases Docs: https://lmstudio.ai/docs/advanced/headless

Perfect! Will give it a run.

Suggestion for the future: When enabling the JIT model load, an option on each model to make them JIT available would be great. I have a ton of models for testing, but users shouldn't necessary be able to run all of them 😄

**Comment by SurenaSKQ** [comment]
I don't know if I should make a separate issue for this, but LM Studio will only have the startup functionality on Windows 11. However, on Windows 10 (LTSC) it doesn't have the functionality.

Also it'd be nice to have this exposed in the settings menu too
